homeassistant/components/sensor/geizhals.py

```diff
@@ -80,6 +80,8 @@ def state(self):
     @property
     def device_state_attributes(self):
         """Return the state attributes."""
+        while (len(self.data.prices) < 4):
+            self.data.prices.append("None")
```

Comment:
It's probably more common to access the attribute directly which wouldn't work when they are varying. I think your change is ok.